samza-core/src/main/java/org/apache/samza/clustermanager/ContainerAllocator.java

```diff
@@ -415,20 +407,34 @@ public void stop() {
 
 
   /**
-   * Checks if a request has expired.
+   * Checks if a request has expired. If this request is due to a control action, then expiry timeout can be overridden
+   * otherwise defaults to
```

Comment:
defaults to ... 
{@link ClusterManagerConfig.CLUSTER_MANAGER_REQUEST_TIMEOUT_MS} if specified or {@link ClusterManagerConfig.DEFAULT_CONTAINER_REQUEST_TIMEOUT_MS}